pkg/render/tiers/tiers.go

```diff
@@ -138,7 +156,56 @@ func (t tiersComponent) allowTigeraClusterDNSPolicy() *v3.NetworkPolicy {
 	}
 }
 
-func createDNSIngressNamespaceSelector(namespaces ...string) string {
+func (t tiersComponent) allowTigeraNodeLocalDNSPolicy() *v3.GlobalNetworkPolicy {
+	nodeLocalDNSPolicySelector := TigeraNamespaceSelector
+
+	nodeLocalDNSPolicy := &v3.GlobalNetworkPolicy{
+		TypeMeta: metav1.TypeMeta{Kind: "GlobalNetworkPolicy", APIVersion: "projectcalico.org/v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: NodeLocalDNSPolicyName,
+		},
+		Spec: v3.GlobalNetworkPolicySpec{
+			Order:    &networkpolicy.AfterHighPrecendenceOrder,
+			Tier:     networkpolicy.TigeraComponentTierName,
+			Selector: nodeLocalDNSPolicySelector,
+			Egress:   []v3.Rule{},
+			Types:    []v3.PolicyType{v3.PolicyTypeEgress},
+		},
+	}
+
+	egressRuleTemplate := v3.Rule{
+		Action: v3.Allow,
+		Destination: v3.EntityRule{
+			// NodeLocal DNSCache creates and listens on the kube-dns ClusterIP on each node, so we can use
+			// kube-dns ClusterIPs address directly in the policy where a normal service IP wouldn't match.
+			Nets:  []string{},
+			Ports: networkpolicy.Ports(53),
+		},
+		Protocol: &networkpolicy.UDPProtocol,
+	}
+
+	if len(t.cfg.DNSEgressCIDRs.IPV4) > 0 {
+		IPV4Rule := egressRuleTemplate
+		appendIPVersionCIDRs(&IPV4Rule.Destination.Nets, t.cfg.DNSEgressCIDRs.IPV4)
+
+		nodeLocalDNSPolicy.Spec.Egress = append(nodeLocalDNSPolicy.Spec.Egress, IPV4Rule)
+	}
+
+	if len(t.cfg.DNSEgressCIDRs.IPV6) > 0 {
+		IPV6Rule := egressRuleTemplate
+		appendIPVersionCIDRs(&IPV6Rule.Destination.Nets, t.cfg.DNSEgressCIDRs.IPV6)
+
+		nodeLocalDNSPolicy.Spec.Egress = append(nodeLocalDNSPolicy.Spec.Egress, IPV6Rule)
+	}
+
+	return nodeLocalDNSPolicy
+}
+
+func appendIPVersionCIDRs(net *[]string, cidrArray []string) {
```

Comment:
I don't think you need this function - you should just be able to do this inline, right?

```go
IPV4Rule.Destination.Nets = append(IPV4.Destination.Nets, t.cfg.DNSEgressCIDRs.IPv4)
```